numpy/random/mtrand/distributions.c

```diff
@@ -729,9 +730,16 @@ long rk_zipf(rk_state *state, double a)
     {
         U = 1.0-rk_double(state);
         V = rk_double(state);
-        X = (long)floor(pow(U, -1.0/am1));
+        X_double = floor(pow(U, -1.0/am1));
+        if ((X_double > (double)LONG_MAX) || (X_double < (double)LONG_MIN))
+        {
```

Comment:
I'd just use `X` without the cast to long and only cast it to long for the return. Note that every place else it is cast float again for the computation and that it is non-negative, so the second check may be avoided. So maybe something like
```
if (X > LONG_MAX) {
    /* X < 1 will be rejected */
    X = 0.0;
    continue;
} 
```
and then put `X < 1` check below first to avoid the big computation. Note that the `continue` above skips the division by `X`, so `0.0` is not a problem.  Also we put the opening brace on the same line as the condition, see `doc/C_STYLE_GUIDE.rst.txt`.

Reply:
Thx for the review! Makes sense, done in my latest commit. We don't need to set X to 0, right? It seems that continue would cause a new X to be computed already. For X<1 we can skip both the big computation as well as the computation in the condition; so I added both conditions in the loop.

Reply:
No, continue goes to the conditional in a do loop.

Reply:
See https://msdn.microsoft.com/en-us/library/6e3dc2z3.aspx.

Reply:
Ah, right. Fixed that.